src/makefiles/linux_x86_64_mkl.mk

```diff
@@ -53,5 +42,8 @@ ifeq ($(findstring clang,$(COMPILER)),clang)
 CXXFLAGS += -Wno-mismatched-tags
 endif
 
-LDFLAGS = $(EXTRA_LDFLAGS) $(OPENFSTLDFLAGS) -rdynamic
-LDLIBS = $(EXTRA_LDLIBS) $(OPENFSTLIBS) $(MKLFLAGS) -lm -lpthread -ldl
+# As late as possible to allow the user to do what they want.
+CXXFLAGS += $(EXTRA_CXXFLAGS)
```

Comment:
the thing with changing might have positive (or negative) impacts in the "active" parameters (IIRC gcc will override the values by the latest one if there are multiple definitions of the same parameters)

Reply:
I know, but this is where we previously violated the (conventional?) semantics of `EXTRA_*FLAGS`  variables. They really should trump everything else. Due to their previous place, I doubt we'd break anyone: override flags had no effect, except to `-O1` which used to came before EXTRA_CXXFLAGS. No one could sensibly put any overrides into this variable in their build scripts anyway. For non-overrides, there is no change.

Reply:
FYI, nvcc will just crash if you give it different -O options like -O1 and later -O2.  (I'm not saying this PR would cause that problem, just FYI).